src/search.c

```diff
@@ -4362,6 +4375,146 @@ typedef struct
 
 #define FUZZY_MATCH_RECURSION_LIMIT	10
 
+// convert tolower simd
+#ifdef PLATFORM_X86
+static inline __m128i simd_tolower(__m128i r)
+{
+    __m128i offset = _mm_add_epi8(r, _mm_set1_epi8(63)); //  A = SCHAR_MIN
+    __m128i cmp = _mm_cmpgt_epi8(_mm_set1_epi8((char)(-102)), offset);
+    __m128i mask = _mm_and_si128(cmp, _mm_set1_epi8(32));
+    return _mm_add_epi8(r, mask);
+}
+#elif defined(PLATFORM_ARM_NEON)
+static inline uint8x16_t simd_tolower(uint8x16_t r UNUSED)
+{
+    uint8x16_t t = vsubq_u8(r, vdupq_n_u8(63));
+    t = vcltq_u8(t, vdupq_n_u8(26));
+    t = vandq_u8(t, vdupq_n_u8(32));
+    return vaddq_u8(r, t);
+}
+#endif
+
+    int
+fuzzy_match_simd(const char_u *pattern, const char_u *text)
```

Comment:
Can you add some comments describing this function and also describe the difference between this function and fuzzy_match()?

Reply:
Experimental draft. So it may look irregular. Please don't mind. I will adjust it when it is ready.